tests/perf-system/analyzer/analyzer.py

```diff
@@ -1,348 +1,490 @@
-        plt.figure()
-        fig, ax = plt.subplots()
-        ax.bar(x_axis, top_counts, 0.9, align="center")
-        ax.set_xticks(x_axis)
-        ax.set_xticklabels(top_bins, rotation=25)
-        plt.ylabel("Count")
-        plt.xlabel("Latency")
-        fig.subplots_adjust(bottom=0.2)
-        plt.savefig("latency_distribution.png")
-
-
-def get_df_from_parquet_file(input_file: str):
-    return pd.read_parquet(input_file)
-
-
-def default_analysis(send_file, response_file):
-    """
-    Produce the analysis results
-    """
-    analysis = Analyze()
-    df_sends = get_df_from_parquet_file(send_file)
-    df_responses = get_df_from_parquet_file(response_file)
-
-    successful_percent = analysis.iter_for_success_and_latency(df_sends, df_responses)
-
-    LOG.info(f"The request type sent is {analysis.get_req_type(df_responses)}")
-
-    print(
-        analysis.time_success_throughput_table(
-            df_sends, df_responses, successful_percent
-        )
-    )
-    print(analysis.latencies_table(df_sends, df_responses))
-
-    x = "-" * 20
-    LOG.info(f'{"".join(x)} Start plotting  {"".join(x)}')
-
-    analysis.plot_latency_by_id(df_sends)
-    analysis.plot_latency_across_time(df_responses)
-    analysis.plot_throughput_per_block(df_responses, 0.1)
-
-    LOG.info(f'{"".join(x)}Finished plotting{"".join(x)}')
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import json
+import re
+import sys
+from email.parser import Parser
+from pathlib import Path
+from typing import List
+
+import matplotlib.pyplot as plt  # type: ignore
+import numpy as np
+import pandas as pd  # type: ignore
+from loguru import logger as LOG
+
+# pylint: disable=import-error
+from prettytable import PrettyTable  # type: ignore
+
+SEC_MS = 1000
+VERBS = {"POST", "GET", "DELETE"}
+DEFAULT_FIGURE_SIZE = (8, 2.5)
+SMALL_FIGURE_SIZE = (4, 2)
+
+# Change default log format
+LOG.remove()
+LOG.add(
+    sys.stdout,
+    format="<green>[{time:HH:mm:ss.SSS}]</green> {message}",
+)
+
+
+class Analyze:
+    def __init__(self):
+        self.latency_list = []
+        self.ms_latency_list = []
+        self.request_verb = []
+        self.throughput = 0
+
+    def get_req_type(self, df_responses: pd.DataFrame) -> str:
+        return df_responses.iloc[0]["rawResponse"].split(b" ")[0].decode("ascii")
+
+    def get_latency_at_i(
+        self, df_sends: pd.DataFrame, df_responses: pd.DataFrame, req_id: int
+    ):
+        # will need to handle the re-submission response from submitter when decided
+        return (
+            df_responses.iloc[req_id]["receiveTime"] - df_sends.iloc[req_id]["sendTime"]
+        )
+
+    def check_success(self, df_responses: pd.DataFrame, req_id: int) -> int:
+        req_resp = df_responses.iloc[req_id]["rawResponse"].split(b"\n")
+        status_list = req_resp[0].split(b" ")
+        # if we get a full statues and says ok increase the successful
```

Comment:
I am worried that this will mark any response matching "\s2[0-9]+[0-9]+\s" as successful, regardless of where the matching takes place...

Reply:
This should match all responses with code 200-299, I recall these were the successful responses (please correct me if I am wrong). However, I don't fully understand: "where the matching takes place"

Reply:
My bad, I had missed the `[0]` when reading the first time, so I was worried it would match something like:

```
HTTP/1.1 400 Bad Request
Content-Type: application/json
Content-Length: some
{
 "msg": "this did not return 200 actually"
}
```